shopinvader_wishlist/services/wishlist.py

```diff
@@ -0,0 +1,291 @@
+# Copyright 2019 Camptocamp (http://www.camptocamp.com).
+# @author Simone Orsi <simone.orsi@camptocamp.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, exceptions
+from odoo.addons.base_rest.components.service import to_int
+from odoo.addons.component.core import Component
+from odoo.osv import expression
+from werkzeug.exceptions import NotFound
+
+
+class WishlistService(Component):
+    _name = "shopinvader.wishlist.service"
+    _inherit = "base.shopinvader.service"
+    _usage = "wishlist"
+    _expose_model = "product.set"
+
+    # The following method are 'public' and can be called from the controller.
+    # All params are untrusted so please check it !
+
+    def get(self, _id):
+        record = self._get(_id)
+        return self._to_json_one(record)
+
+    def search(self, **params):
+        return self._paginate_search(**params)
+
+    # pylint: disable=W8106
+    def create(self, **params):
+        if not self._is_logged_in():
+            # TODO: is there any way to control this in the REST API?
+            raise exceptions.UserError(
+                _("Must be authenticated to create a wishlist")
+            )
+        vals = self._prepare_params(params.copy())
+        record = self.env[self._expose_model].create(vals)
+        self._post_create(record)
+        return {"data": self._to_json_one(record)}
+
+    def update(self, _id, **params):
+        record = self._get(_id)
+        record.write(self._prepare_params(params.copy(), mode="update"))
+        self._post_update(record)
+        return self.search()
+
+    def delete(self, _id):
+        self._get(_id).unlink()
+        return self.search()
+
+    def add_to_cart(self, _id):
+        record = self._get(_id)
+        cart = self.component(usage="cart")._get()
+        wizard = self.env["product.set.add"].create(
+            {
+                "order_id": cart.id,
+                "partner_id": self.partner_user.id,
+                "product_set_id": record.id,
+                "skip_existing_products": True,
+            }
+        )
+        wizard.add_set()
+        res = self.search()
+        # invalidate cart cache
+        res["store_cache"] = {"cart": {}}
```

Comment:
```suggestion
        res["store_cache"] = {"cart": self.component(usage="cart")._convert_one_sale(cart)}
```

Reply:
@simahawk If you push an empty cart into the cache it will not remove it from the cache but set an empty cart. This is why we do not see any cart on the front and we need to add a product to refresh it.
We need to review the cache management to make it more human

Reply:
@sebastienbeau but the cart should be already in the session, I wanted only to invalidate it's cache.... but I see your point, I'll try! Thanks!

Reply:
I did it a little bit different but it works now! Thanks for the pointer :)